File: python/src/acp_sdk/models/platform.py
from enum import Enum

from pydantic import BaseModel, ConfigDict


class PlatformUIType(str, Enum):
    CHAT = "chat"
    HANDSOFF = "hands-off"

class ConfigurationField(str, Enum):
    BOOLEAN = "boolean"


class AgentToolInfo(BaseModel):
    name: str
    description: str | None = None
    model_config = ConfigDict(extra="allow")


class AgentConfiguration(BaseModel):
    name: str
    description: str | None = None
    type: ConfigurationField


class PlatformUIAnnotation(BaseModel):
    ui_type: PlatformUIType
    user_greeting: str | None = None
    display_name: str | None = None
    configuration: list[AgentConfiguration] = []
    tools: list[AgentToolInfo] = []
    model_config = ConfigDict(extra="allow")

File: typescript/src/models/platform.ts
import { z } from "zod";

export const PlatformUIType = z.enum(["chat", "hands-off"]);

export const AgentToolInfo = z.object({
  name: z.string(),
  description: z.string().nullish(),
});

export const PlatformUIAnnotation = z.object({
  ui_type: PlatformUIType,
  user_greeting: z.string().nullish(),
  display_name: z.string().nullish(),
  tools: z.array(AgentToolInfo).default([]),
});
